tests/test_modeling_gpt2.py

```diff
@@ -176,6 +182,19 @@ def create_and_check_lm_head_model(self, config, input_ids, input_mask, head_mas
                 list(result["lm_logits"].size()), [self.batch_size, self.seq_length, self.vocab_size]
             )
 
+        def create_and_check_generate_lm_head_model(
```

Comment:
This kind of language generation test (using random weights and only checking that no error is thrown) should be relatively cheap and could be added for each test_modeling_<lm_model> file individually. 
What do you think? @LysandreJik and @sshleifer

Reply:
Those tests would be a good addition, indeed!

Reply:
20 forward passes for the model and since it's a dummy model it should be pretty fast

Reply:
I measured how the tests increased the test time locally on my machine. Here the result:

| LM Model  | Increase in test time |
| ------------- | ------------- |
| XLNet  | 8.0s -> 9.7s |
| XLNet  | 8.0s -> 9.7s |
| XLNet  | 8.0s -> 9.7s |

Reply:
At the moment, I included tests that cover almost all the different options to generate language: 

- no_beam_search_decoding & sampling
- no_beam_search_decoding & greedy decoding
- beam_search_decoding & sampling
- beam_seach_decoding & greedy decoding
- batch_size = 1 vs. batch_size > 1
- input_ids = None vs. input_ids set by user

If the test take too much time I can try to make them more efficient and/or delete some cases
@LysandreJik, @sshleifer